src/parser/parser-utils.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+const ParserUtils = {
+  BufferToStr: (buffer: ArrayBuffer): string => {
+    const textDecoder = new TextDecoder();
+    return textDecoder.decode(buffer);
+  },
+  StrToBuffer: (str: string): Uint8Array => {
+    const textEncoder = new TextEncoder();
+    return textEncoder.encode(str);
+  }
+};
+
+const MpdUtils = {
+  TagTypes: {
+    SEGMENT_TEMPLATE: 'SegmentTemplate',
+    REPRESENTATION: 'Representation',
+    ESSENTIAL_PROPERTY: 'EssentialProperty',
+    ADAPTATION_SET: 'AdaptationSet'
+  },
+  AttributeTypes: {
+    ID: 'id',
+    CONTENT_TYPE: 'contentType',
+    MIME_TYPE: 'mimeType',
+    MEDIA: 'media',
+    DURATION: 'duration',
+    START_NUMBER: 'startNumber',
+    TIMESCALE: 'timescale',
+    PRESENTATION_TIME_OFFSET: 'presentationTimeOffset',
+    END_NUMBER: 'endNumber',
+    BANDWIDTH: 'bandwidth',
+    WIDTH: 'width',
+    HEIGHT: 'height',
+    SCHEME_ID_URI: 'schemeIdUri',
+    VALUE: 'value'
+  }
+};
+
+const XmlUtils = {
+  parseXml: (text: string): Document => {
+    const domParser = new DOMParser();
+    return domParser.parseFromString(text, 'text/xml');
+  },
+  parseAttr: (elem: HTMLElement, name: string, parseFunction?: Function, defaultValue: any): any => {
+    let parsedValue = null;
+    const value = elem.getAttribute(name);
+    if (value !== null) {
+      parsedValue = parseFunction ? parseFunction(value) : value;
+    }
+    return parsedValue == null ? defaultValue : parsedValue;
```

Comment:
dont see `defaultValue` usage is it for the future?

Reply:
yes